### DISCUSSION TITLE
Visible Statistics

### DISCUSSION THREAD

**Discussion by UpYoursMicrosoft** [nroi/flexo#119] [Ideas & Feature Requests]
Flexo has saved lots of my own time, bandwidth, and, more importantly, that of my local Arch mirrors. However, I'd like to be able to see the exact cache hit ratio and total bandwidth saved. I've hacked together a couple of different bash one-liners to do that by parsing the logs, but I'd really like to see that as part of Flexo.

Something as simple as a periodic dump of those statistics in the logs would be great, but it would be even better if it exported Prometheus metrics since there's already a Web server running.

I'm not sure if any of this is practical with the current implementation but I think it would be a real improvement.

Thank you.

**Comment by nroi** [comment]
Thanks for the feedback and the suggestion.
I see how something like this could be useful, but at the moment, I have only a very limited amount of time available for Flexo, and the first priority is to fix bugs and improve the maintainability (e.g., by improving the code quality). So, it's unlikely that I'm going to implement this feature in the short term or midterm.

**Comment by UpYoursMicrosoft** [comment]
Thanks for responding. I completely understand. I just wanted to get the idea out there. I would add the feature myself if I could but I'm really not much of a programmer and I've never even touched Rust.
